### PR TITLE
Post status filter

### PR DESCRIPTION
**Pull request by ataylorme**
* Add get_post_types and get_post_statuses helper methods; Addresses #229
* Remove unnecessary get_env call
* Add solr_post_status filter
* Add missing solr_post_types filter to get_post_types call
* Use PANTHEON_INDEX_PORT instead of hard-coded port in curl

### CODE REVIEW

File: includes/class-solrpower-api.php [hunk 1]
```diff
@@ -432,7 +432,7 @@ function index_stats() {
 		$stats     = wp_cache_get( $cache_key, 'solr' );
 		if ( false === $stats ) {
 
-			$post_types = get_post_types( array( 'exclude_from_search' => false ) );
+			$post_types = apply_filters('solr_post_types', get_post_types( array( 'exclude_from_search' => false ) ) );
```
Comment (danielbachhuber) [inline includes/class-solrpower-api.php]: Should this use the filter too?

File: includes/class-solrpower.php [hunk 2]
```diff
@@ -313,4 +313,22 @@ function ajax_search() {
...
+	}
+
+	/**
+	 * Returns the post statuses used for Solr after applying the solr_post_status filter
+	 *
+	 * @return array
+	 */
+	public function get_post_statuses() {
+		return apply_filters( 'solr_post_status', array( 'publish' ) );
+	}
```
Comment (danielbachhuber) [inline includes/class-solrpower.php]: Can we make these static methods, so they don't need to be called with the instance?

Comment (ataylorme) [inline includes/class-solrpower.php]: Good call, updated.

File: includes/class-solrpower-api.php (hunk 1 above)
Comment (ataylorme) [inline includes/class-solrpower-api.php]: 👍

**Comment by ataylorme** [comment]
@danielbachhuber can you review again?

**Comment by gtenaschuk** [comment]
Great! Going to update my repo.
I was not available today to make the changes, thanks guys!

**Comment by ataylorme** [comment]
Thank you for submitting the PR to begin with!

**Pull request merged**